lib/suite.js

```diff
@@ -66,7 +67,7 @@ function Suite(title, parentContext) {
   this._beforeAll = [];
   this._afterEach = [];
   this._afterAll = [];
-  this.root = !title;
+  this.root = root || false;
```

Comment:
Prefer insurance myself...
```js
   this.root = isRoot === true;
```

Reply:
@plroebuck thanks, an alternative way could be:
1. empty title "" results in `root = true` (no changes so far)
2. in [Suite.create](https://github.com/mochajs/mocha/blob/master/lib/suite.js#L33): test for empty title and existing parent
```
exports.create = function(parent, title) {
  var suite = new Suite(title, parent.ctx);
  suite.parent = parent;
  if (!title && parent) {       <<< insert
    suite.root = false;         <<< insert
  }                             <<< insert
  title = suite.fullTitle();    // ???
  parent.addSuite(suite);
  return suite;
};
```
It would still be possible to manually create a suite-tree with several roots via `Suite.addSuite`, f.e. in external interfaces or test fixtures. As per now I would not prevent this, but could move `if (!title && parent) {...}` to `Suite.addSuite`.

Do you prefer this alternative solution?

Reply:
Don't see how `Suite.create` would be helpful as "lib/mocha.js" invokes the `Suite` ctor directly. `Context` ctor from "lib/context.js" is _only_ used by `Mocha` ctor; if "flag" set there, then Mocha ctor could designate root suite based on it.

Reply:
Mocha constructor is not changed in my last proposition, just `Suite.create`.

Reply:
If it's still possible to create multiple root suites, it's still broken -- [there can be only one](https://www.youtube.com/watch?v=sqcLjcSloXs) -- the suite created in the `Mocha` ctor!

Reply:
ok, done.
I added a line in `Suite.prototype.addSuite` which is actually the only change required for this PR.